hive/lib/src/backend/stub/storage_backend_memory.dart

```diff
@@ -0,0 +1,127 @@
+import 'dart:typed_data';
+
+import 'package:hive/hive.dart';
+import 'package:hive/src/backend/storage_backend.dart';
+import 'package:hive/src/backend/vm/read_write_sync.dart';
+import 'package:hive/src/binary/binary_reader_impl.dart';
+import 'package:hive/src/binary/binary_writer_impl.dart';
+import 'package:hive/src/binary/frame.dart';
+import 'package:hive/src/binary/frame_helper.dart';
+import 'package:hive/src/box/keystore.dart';
+
+/// In-memory Storage backend
+class StorageBackendMemory extends StorageBackend {
+  final HiveCipher? _cipher;
+
+  final ReadWriteSync _sync;
+
+  final FrameHelper _frameHelper;
+
+  Uint8List _bytes;
+
+  TypeRegistry? _typeRegistry;
+
+  int _writeOffset = 0;
```

Comment:
I just had the fancy idea of creating an abstracted layer for this:

At the moment, the `StorageBackendMemory` contains both the `_bytes` and the `_writeOffset` property. What about unifying them to a single property? I'd recommend e.g.:

```dart
class RandomAccessBuffer extends Uint8List {
    int _writeOffset;
    
    RandomAccessBuffer(Uint8List bytes, {int initialOffset = 0});
}
```

- should provide a constructor from a `Uint8List` and the initial offset
- have ready-to-use `write()`, `read()` and `movePointer()` methods